sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/models/DigitalTwinModelData.java

```diff
@@ -0,0 +1,199 @@
+package com.azure.digitaltwins.core.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.digitaltwins.core.implementation.models.ModelData;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.time.OffsetDateTime;
+import java.util.Map;
+
+/**
+ * The DigitalTwinModelData representing the model and its corresponding metadata.
+ */
+@Fluent
+public class DigitalTwinModelData {
+
+    private final ObjectMapper mapper;
+
+    /**
+     * Creates a new instance of DigitalTwinModelData.
+     * @param modelData The model data.
+     */
+    public DigitalTwinModelData(ModelData modelData) {
```

Comment:
This constructor is currently public because this class lives in a different package than the convenience layer.

Reply:
A user should never need to initialize this type, correct?

Reply:
correct, I am talking to Srikanta to see what we should do here.

Reply:
@srnagar

Reply:
Spoke to @azabbasi offline and discussed 3 options.
- Leave the public ctor as-is since this is output-only model and the user is not expected to create a new instance
- Use an interface for public model type and define a concrete type in implementation package
- Use a helper type to convert from internal type to public type and make the ctor package-private on the public type. 

For options 1 and 3, flatten the internal ModelData type and pass the properties directly.

Reply:
Thank you Srikanta. I went with the 3rd option as it seems to be the cleanest